futurex_openedx_extensions/helpers/extractors.py

```diff
@@ -2,27 +2,133 @@
         if result:
             return result.groupdict().get('course_id')
 
     return None
 
 
 def get_first_not_empty_item(items: List, default: Any = None) -> Any:
-    """Return the first item in the list that is not empty."""
+    """
+    Return the first item in the list that is not empty.
+
+    :param items: List of items to check.
+    :type items: List
+    :param default: Default value to return if no item is found.
+    :type default: Any
+    :return: First item that is not empty.
+    :rtype: Any
+    """
     return next((item for item in items if item), default)
+
+
+def verify_course_ids(course_ids: List[str]) -> None:
+    """
+    Verify that all course IDs in the list are in valid format. Raise an error if any course ID is invalid.
+
+    :param course_ids: List of course IDs to verify.
+    :type course_ids: List[str]
+    """
+    if course_ids is None:
+        raise ValueError('course_ids must be a list of course IDs, but got None')
+
+    for course_id in course_ids:
+        if not isinstance(course_id, str):
+            raise ValueError(f'course_id must be a string, but got {type(course_id).__name__}')
+        if not re.search(COURSE_ID_REGX_EXACT, course_id):
+            raise ValueError(f'Invalid course ID format: {course_id}')
+
+
+def get_orgs_of_courses(course_ids: List[str]) -> Dict[str, Any]:
+    """
+    Get the organization of the courses with the given course IDs.
+
+    :param course_ids: List of course IDs to get the organization of.
+    :type course_ids: List[str]
+    :return: Dictionary containing the organization of each course ID.
+    :rtype: Dict[str, Any]
+    """
+    verify_course_ids(course_ids)
+    courses = CourseOverview.objects.filter(id__in=course_ids)
+
+    result: Dict[str, Any] = {
+        'courses': {str(courses.id): courses.org.lower() for courses in courses},
+    }
+    result['invalid_course_ids'] = [course_id for course_id in course_ids if course_id not in result['courses']]
+    return result
+
+
+def generate_simple_hashcode(
+    value_dict: Dict[str, Any],
+    fields: List[str],
+    separator: str = ',',
+    replace_none: str = 'None',
+) -> str:
+    """
+    Generate a hashcode string from the given dictionary. The dictionary must be a simple Dict[str: Any], typically
+    like the result of using the values() method of a queryset.
+
+    {role: 'admin', org: 'ORG1', course_id: 'COURSE1'} ====> 'admin,ORG1,COURSE1'
+
+    :param value_dict: Dictionary to generate hashcode string from.
+    :type value_dict: Dict[str, Any]
+    :param fields: List of field names to include in the hashcode.
+    :type fields: List[str]
+    :param separator: Separator to use between field values.
+    :type separator: str
+    :param replace_none: String to replace None values with.
+    :type replace_none: str
+    :return: Hashcode string.
+    """
+    return separator.join(replace_none if value_dict[field] is None else str(value_dict[field]) for field in fields)
```

Comment:
This way the returned code can both be used as a Dict key _and_ as an information holder without needing to parse strings.

```suggestion
    return Hashcode(role, org, course_id)
```

Reply:
Done @OmarIthawi , please check